app/src/main/res/values/strings.xml

```diff
@@ -315,4 +316,26 @@
     <string name="download_confirmation_metered">You are currently on a metered connection, and data charges from your service provider may apply.\n\nDo you still want to continue?</string>
     <string name="download_update_confirmation">Download update?</string>
     <string name="no_contributors_found">No contributors found</string>
+
+    <string name="installation_failed">Installation failed</string>
+    <string name="installation_cancelled">Installation cancelled</string>
+    <string name="installation_blocked">Installation blocked</string>
+    <string name="installation_conflict">Installation conflict</string>
+    <string name="installation_incompatible">Installation incompatible</string>
+    <string name="installation_invalid">Installation invalid</string>
+    <string name="installation_storage_issue">Installation storage issue</string>
+    <string name="installation_timeout">Installation timeout</string>
+    <string name="installation_success">Installation succeeded</string>
+
+    <string name="installation_failed_description">The installation failed due to an unknown reason. Please try again.</string>
+    <string name="installation_aborted_description">The installation was cancelled manually.</string>
+    <string name="installation_blocked_description">The installation was blocked. Adjust your security settings and try again.</string>
+    <string name="installation_conflict_description">An existing installation of the app prevents the installation. Uninstall the app and try again?</string>
+    <string name="installation_incompatible_description">The app is incompatible with this device. Contact the developer of the app and ask for support.</string>
+    <string name="installation_invalid_description">The app is invalid. Uninstall the app and try again?</string>
+    <string name="installation_storage_issue_description">The app could not be installed due to insufficient storage. Free up some space and try again.</string>
+    <string name="installation_timeout_description">The installation took too long.</string>
+    <string name="installation_success_description">The app has been installed successfully.</string>
```

Comment:
Didn't we say that these should be changed to something like "action" instead of "installation"?

Reply:
The strings refer to the entire installation activity that includes uninstallation as an action, and hence, if we need to uninstall to be able to install, but the uninstallation action fails, the entire activity is terminated, and the strings are displayed accordingly.

Reply:
Right. I forgot about that